cfme/common/provider.py

```diff
@@ -130,6 +130,14 @@ def __init__(self, **kwargs):
             if self.type == 'service_account':
                 self.service_account = kwargs['service_account']
 
+    def __hash__(self):
+        return hash(self.key) ^ hash(type(self))
```

Comment:
`it is advised to somehow mix together (e.g. using exclusive or) the hash values for the components of the object that also play a part in comparison of objects.`